File: OOPs/super Keyword/super.py
class Parent:
    def __init__(self, value):
        self.parent_value = value
        print(f"Parent constructor called with value: {self.parent_value}")
    
    def display(self):
        print(f"Parent value: {self.parent_value}")

class Child(Parent):
    def __init__(self, value, child_value):
        super().__init__(value)  # Call Parent constructor
        self.child_value = child_value
        print(f"Child constructor called with value: {self.child_value}")
    
    def display(self):
        super().display()  # Call Parent display method


obj = Child(100, 200)
obj.display()

File: OOPs/super Keyword/Super.java
class Parent {
    int parentValue;

    Parent(int value) {
        this.parentValue = value;
        System.out.println("Parent Constructor called with value: " + value);
    }
    void display() {
        System.out.println("Parent's display method");
    }
}

class Child extends Parent {
    Child(int value) {
        super(value);  // Call Parent class constructor
        System.out.println("Child Constructor called");
    }
    public void display() {
        super.display();  // Call Parent class method
        System.out.println("Value of parentValue: " + parentValue);
    }
}

public class Super {
    public static void main(String[] args) {
        Child obj = new Child(100);
        obj.display();
    }
}
